types/src/on_chain_config.rs

```diff
@@ -9,7 +9,38 @@ use crate::{
 };
 use libra_crypto::HashValue;
 use move_core_types::identifier::Identifier;
-use serde::{Deserialize, Serialize};
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+
+/// Trait to be implemented by a storage type from which to read on-chain configs
+#[async_trait::async_trait(?Send)]
+pub trait ConfigStorage {
+    async fn fetch_config(&self, access_path: AccessPath) -> Option<Vec<u8>>;
+}
+
+/// Trait to be implemented by a Rust struct representation of an on-chain config
+/// that is stored in storage as a deserialized byte array
+#[async_trait::async_trait(?Send)]
+pub trait OnChainConfig {
+    const IDENTIFIER: &'static str;
+
+    async fn fetch_config<U: ConfigStorage>(storage: &U) -> anyhow::Result<Self>
```

Comment:
I wonder should we split this method into two
1) that builds access_path from identifier
2) that recreates object from bytes
in such way:
- in main state sync routine you can batch all read requests in single call
- you can avoid using async_trait and tokio dependency in libra-vm

Reply:
I do believe that we should try to avoid bringing in the  async_trait and tokio dependency in both libra-vm and libra-types.

Reply:
I don't say that's a primary reason, but that's just unnecessary
IMO split makes flow more efficient and cleaner. Also VmConfig:: load_on_chain_config looks weird now. tokio runtime is only used there to block on single function

Reply:
I think I found a way to avoid async_trait and tokio dependencies, will update in next push. re: batching read requests, I think that's a great idea, but I think I'll implement it once in follow-up diff where I implement actual use case in state sync - I'll have a better sense of what the interface should be if I play around with the actual code 😄